chronosphere/resource_notification_policy_independent.go

```diff
@@ -138,13 +140,33 @@ func notificationRoutesFromModel(
 			Severity:       sev,
 			Notifiers:      sliceutil.Map(f.NotifierSlugs, tfid.Slug),
 			RepeatInterval: durationFromSecs(f.RepeatIntervalSecs),
+			GroupBy:        notificationRouteFromGroupByModel(f),
 		})
 	}
 	load(warn, m.Warn)
 	load(critical, m.Critical)
 	return out
 }
 
+func notificationRouteToGroupByModel(r intschema.NotificationRoute) *models.NotificationPolicyRoutesGroupBy {
+	if r.GroupBy == nil {
+		return nil
+	}
+
+	return &models.NotificationPolicyRoutesGroupBy{
+		LabelNames: r.GroupBy.LabelNames,
+	}
+}
+
+func notificationRouteFromGroupByModel(f *models.RoutesNotifierList) *intschema.NotificationRouteGroupBy {
+	if f == nil || f.GroupBy == nil {
+		return nil
+	}
```

Comment:
if `f == nil` is already handled in the caller, and this method should only use `models.NotificationPolicyRoutesGroupBy`